tasks/ServiceFabricDockerPush/containerpush.ts

```diff
@@ -0,0 +1,297 @@
+        // Generate a chain of promises that sequentially push the docker images
+        let pushDockerImagesPromise = taggedImages.reduce((pushPromise: Q.Promise<void[]>, image) => {
+            return pushPromise.then(() => {
+                return dockerPush(connection, image.taggedName).then(imageDigest => imageDigests.push(imageDigest));
+            });
+        }, Q.resolve(null));
+        
+        // Wait for the Docker images to be pushed to the registry and then update the service manifests.
+        return pushDockerImagesPromise.then(() => {
+            return updateServiceManifests(taggedImages, imageDigests, applicationPackagePath);
+        });
+    });
+
+    return <Q.Promise<void>><any>runPromise;
+}
+
+/**
+ * Asynchronously updates the service manifest files so that they reference the image digests associated with the pushed images.
+ * @param taggedImages Full set of images that have been pushed.
+ * @param imageDigests Image digest values for each of the images that have been pushed.
+ * @param applicationPackagePath Path to the Service Fabric application package.
+ */
+function updateServiceManifests(taggedImages: ImageNameInfo[], imageDigests: string[], applicationPackagePath: string): Q.Promise<void[]> {
+    let serviceManifestPaths = fs.readdirSync(applicationPackagePath)
+        .filter(f => fs.statSync(path.join(applicationPackagePath, f)).isDirectory())
+        .map(d => path.join(applicationPackagePath, d, "ServiceManifest.xml"));
+
+    // The index of each service manifest corresponds to the same index within the serviceManifestPaths array.
+    let serviceManifests = [];
+
+    // Generate a chain of promises that sequentially read the service manifest files
+    let readServiceManifestsPromise = serviceManifestPaths.reduce((readManifestPromise: Q.Promise<any>, serviceManifestPath) => {
+        return readManifestPromise.then(() => {
+            return readXmlFileAsJson(serviceManifestPath).then(json => serviceManifests.push(json) );
+        });
+    }, Q.resolve(null));
+
+    let serviceManifestSavePromises: Q.Promise<void>[] = [];
+
+    // Wait for the service manifest files to be read
+    return readServiceManifestsPromise.then(() => {
+        let imageNameToServiceManifestMapping = getImageNameToServiceManifestInfoMapping(serviceManifests, serviceManifestPaths);
+
+        // Go through each of the images that were pushed and find those that were tagged with latest.
+        for (let imageIndex = 0; imageIndex < taggedImages.length; imageIndex++) {
+            let image = taggedImages[imageIndex];
+            if (!image.hasLatestTag) {
+                continue;
+            }
+
+            // Find the service manifest which is configured to use this image
+            let serviceManifestImageNameInfo = imageNameToServiceManifestMapping[image.originalName];
+            if (serviceManifestImageNameInfo) {
+                let newImageName = `${image.taggedName}@${imageDigests[imageIndex]}`;
```

Comment:
Also you need to qualify the beginning of the name with the registry. See qualifyImageName in containerconnection.ts